### PR TITLE
refactor(profiling): use envier for configuration

### PR DESCRIPTION
**Pull request by P403n1x87**
We consolidate the profiler configuration and documentation using envier.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] PR description includes explicit acknowledgement/acceptance of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/profiling/collector/__init__.py [hunk 1]
```diff
@@ -77,5 +77,5 @@ def _create_capture_sampler(collector):
 
 @attr.s
 class CaptureSamplerCollector(Collector):
-    capture_pct = attr.ib(factory=attr_utils.from_env("DD_PROFILING_CAPTURE_PCT", 1.0, float))
+    capture_pct = attr.ib(type=float, default=config.capture_pct)
```
Comment (sanchda) [inline ddtrace/profiling/collector/__init__.py:80]: Love how this helps us be more structured about envvars

**Review by sanchda — APPROVED** [review]
This is really good stuff.  I have two wishes--they're not even complaints or nits, just reflections on the state of the universe.

1.  I wish we were able to preserve the literal name of the environment variable.  On one hand, it's nice to get a guarantee that the keyname of the environment variable is expanded in a consistent way from the name of the configuration, but for folks who are trying to grep the codebase for where environment variables are set/referred, it's a bit annoying.  I wonder if there's value in adopting the stylistic standard of annotating a config with the underlying environment variable.  That seems like a weird, non-idiomatic burden.

2. I like how default values are collected, but at the same time if a configuration is only consumed in one place, this delocalizes the declaration of the default from the (only) location in which it is consumed.  I _think_ it's probably better to have it this way, since a centralized representation of default values has its advantages.  Ultimately, a small price to pay, I think.


LVGTM, thank you!

**Review by emmettbutler — APPROVED** [review]
This is awesome. Have you verified that the environment variable documentation in `configuration.rst` on this branch is the same as on master?

**Comment by P403n1x87** [comment]
image

**Pull request merged**